### ISSUE TITLE
Add support for Scala Native

### ISSUE THREAD

**Issue by djneades** [scalatest/scalatest#1112]
It is early days for Scala Native (https://github.com/scala-native/scala-native), but support would be be very helpful.

**Comment by xuwei-k** [comment]
related to https://github.com/scala-native/scala-native/issues/339

**Comment by bvenners** [comment]
Would love to support Scala Native. At Scala eXchange (I think it was) we did get Scalactic to compile under Scala native. ScalaTest had a lot more issues at the time.

**Comment by darkfrog26** [comment]
No Scala Native, no Scala.js 2.13.0-M1 support...it's stuff like this that make me think perhaps Specs2 is a better testing solution.

**Comment by bvenners** [comment]
@darkfrog26 when we did the 2.13.0-M1 build, scalacheck wasn't available for Scala.js, so we couldn't build ScalaTest for Scala.js 2.13.0-M1. Looking now it seems we are still blocked by that.

As far as Scala Native goes, we have on two occasions tried that, The first time Scala Native just wasn't far enough along. The second time we got Scalactic working under Scala Native, but didn't have time to go further. It too us a year to support Scala.js, and I'm sure Scala Native won't nearly be that bad, but it may take some time, and we've been focused on other things. I do definitely want to support Scala Native and dotty as well, but we haven't done either of those yet.

**Comment by darkfrog26** [comment]
@bvenners I'd like to apologize for my earlier comment. I was having a very frustrating time trying to get my code updated to work with Scala Native only to have ScalaTest be the roadblock for the second time (last time being trying to add 2.13 support).  I know you all work hard and do great work and don't deserve being attacked for not supporting absolutely everything.

**Comment by xuwei-k** [comment]
https://repo1.maven.org/maven2/org/scalatest/scalatest_native0.3_2.11/

`3.2.0-SNAP10` support scala-native?

**Comment by shadaj** [comment]
@darkfrog26 are you running on JVM? `NoClassDefFoundError`s don't exist in Scala Native.

Also, ScalaCheck does support Scala Native and it's combination with ScalaTest is being used successfully in other projects.

**Comment by darkfrog26** [comment]
At the moment I'm just testing JVM to make sure JVM and JS work before I attempt adding Native.

I was searching Maven Central and didn't see any native dependencies there...

**Comment by shadaj** [comment]
@darkfrog26 in that case, this may be a bug with ScalaTest JVM loading ScalaCheck too early. Can you verify that your code is not using any of the ScalaCheck support functions and file a separate issue for this?

**Comment by darkfrog26** [comment]
Okay, I'll dig in a bit further, but here's the complete stack trace:

```
[error] java.lang.NoClassDefFoundError: org/scalacheck/Test$TestCallback
[error]         at java.lang.Class.getDeclaredFields0(Native Method)
[error]         at java.lang.Class.privateGetDeclaredFields(Class.java:2583)
[error]         at java.lang.Class.getDeclaredFields(Class.java:1916)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror$FromJavaClassCompleter.$anonfun$completeRest$4(JavaMirrors.scala:787)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror$FromJavaClassCompleter.$anonfun$completeRest$1(JavaMirrors.scala:797)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror$FromJavaClassCompleter.completeRest(JavaMirrors.scala:755)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror$FromJavaClassCompleter.complete(JavaMirrors.scala:751)
[error]         at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1530)
[error]         at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anon$9.scala$reflect$runtime$SynchronizedSymbols$SynchronizedSymbol$$super$info(SynchronizedSymbols.scala:191)
[error]         at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol.$anonfun$info$1(SynchronizedSymbols.scala:129)
[error]         at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol.info(SynchronizedSymbols.scala:125)
[error]         at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol.info$(SynchronizedSymbols.scala:129)
[error]         at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anon$9.info(SynchronizedSymbols.scala:191)
[error]         at scala.reflect.internal.Symbols$Symbol.initialize(Symbols.scala:1678)
[error]         at scala.reflect.internal.Symbols$Symbol.hasFlag(Symbols.scala:738)
[error]         at scala.reflect.internal.Symbols$Symbol.hasFlag(Symbols.scala:741)
[error]         at scala.reflect.internal.Symbols$Symbol.isStatic(Symbols.scala:993)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror.reflectModule(JavaMirrors.scala:189)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror.reflectModule(JavaMirrors.scala:55)
[error]         at org.scalatest.tools.Framework.runner(Framework.scala:1021)
[error]         at sbt.Defaults$.$anonfun$createTestRunners$1(Defaults.scala:854)
[error]         at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[error]         at scala.collection.immutable.Map$Map1.foreach(Map.scala:120)
[error]         at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[error]         at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[error]         at scala.collection.AbstractTraversable.map(Traversable.scala:104)
[error]         at sbt.Defaults$.createTestRunners(Defaults.scala:848)
[error]         at sbt.Defaults$.allTestGroupsTask(Defaults.scala:906)
[error]         at sbt.Defaults$.$anonfun$testTasks$10(Defaults.scala:676)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:39)
[error]         at sbt.std.Transform$$anon$4.work(System.scala:66)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:262)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[error]         at sbt.Execute.work(Execute.scala:271)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:262)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:174)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:36)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]         at java.lang.Thread.run(Thread.java:748)
[error] Caused by: java.lang.ClassNotFoundException: org.scalacheck.Test$TestCallback
[error]         at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
[error]         at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[error]         at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[error]         at java.lang.Class.getDeclaredFields0(Native Method)
[error]         at java.lang.Class.privateGetDeclaredFields(Class.java:2583)
[error]         at java.lang.Class.getDeclaredFields(Class.java:1916)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror$FromJavaClassCompleter.$anonfun$completeRest$4(JavaMirrors.scala:787)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror$FromJavaClassCompleter.$anonfun$completeRest$1(JavaMirrors.scala:797)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror$FromJavaClassCompleter.completeRest(JavaMirrors.scala:755)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror$FromJavaClassCompleter.complete(JavaMirrors.scala:751)
[error]         at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1530)
[error]         at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anon$9.scala$reflect$runtime$SynchronizedSymbols$SynchronizedSymbol$$super$info(SynchronizedSymbols.scala:191)
[error]         at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol.$anonfun$info$1(SynchronizedSymbols.scala:129)
[error]         at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol.info(SynchronizedSymbols.scala:125)
[error]         at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol.info$(SynchronizedSymbols.scala:129)
[error]         at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anon$9.info(SynchronizedSymbols.scala:191)
[error]         at scala.reflect.internal.Symbols$Symbol.initialize(Symbols.scala:1678)
[error]         at scala.reflect.internal.Symbols$Symbol.hasFlag(Symbols.scala:738)
[error]         at scala.reflect.internal.Symbols$Symbol.hasFlag(Symbols.scala:741)
[error]         at scala.reflect.internal.Symbols$Symbol.isStatic(Symbols.scala:993)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror.reflectModule(JavaMirrors.scala:189)
[error]         at scala.reflect.runtime.JavaMirrors$JavaMirror.reflectModule(JavaMirrors.scala:55)
[error]         at org.scalatest.tools.Framework.runner(Framework.scala:1021)
[error]         at sbt.Defaults$.$anonfun$createTestRunners$1(Defaults.scala:854)
[error]         at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[error]         at scala.collection.immutable.Map$Map1.foreach(Map.scala:120)
[error]         at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[error]         at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[error]         at scala.collection.AbstractTraversable.map(Traversable.scala:104)
[error]         at sbt.Defaults$.createTestRunners(Defaults.scala:848)
[error]         at sbt.Defaults$.allTestGroupsTask(Defaults.scala:906)
[error]         at sbt.Defaults$.$anonfun$testTasks$10(Defaults.scala:676)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:39)
[error]         at sbt.std.Transform$$anon$4.work(System.scala:66)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:262)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[error]         at sbt.Execute.work(Execute.scala:271)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:262)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:174)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:36)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]         at java.lang.Thread.run(Thread.java:748)
[error] (Test / executeTests) java.lang.NoClassDefFoundError: org/scalacheck/Test$TestCallback
```

**Comment by pjazdzewski1990** [comment]
Any luck with solving the issue? I'm observing a similar `java.lang.NoClassDefFoundError: org/scalacheck/Test$TestCallback` when on `"3.2.0-SNAP10"`. 
It has been mentioned that it could be a scalatest issue, if so is there a GH issue I could follow (or maybe should I create one)?

**Comment by darkfrog26** [comment]
@pjazdzewski1990, take a look at https://github.com/outr/reactify where I got JVM, JS, and Native all building and testing properly

**Comment by jaceklaskowski** [comment]
Looks like an issue with ScalaTest. You have to add ScalaCheck to the dependencies as follows:

```
libraryDependencies += "org.scalatest" %% "scalatest" % "3.2.0-SNAP10" % Test
// scalacheck to fix an issue with scalatest
// (Test / executeTests) java.lang.NoClassDefFoundError: org/scalacheck/Test$TestCallback
libraryDependencies += "org.scalacheck" %% "scalacheck" % "1.14.0" % Test
```

**Comment by prkapoor** [comment]
@jaceklaskowski, thank you for the solution. Which file do I put this configuration in. I have a maven pom

**Comment by lolgab** [comment]
I've published the artifact under my groupid having seen no interest about Scala Native by scalacheck maintainer. You can use `"com.github.lolgab" %%% "scalacheck" % "0.4.1"` as Scala Native scalacheck dependency.

**Comment by SethTisue** [comment]
ScalaCheck maintenance is more active now, and artifacts are now available for Scala Native

**Comment by tindzk** [comment]
I believe this issue can be closed given that ScalaTest now has [official Scala Native artefacts](https://repo1.maven.org/maven2/org/scalatest/scalatest_native0.4.0-M2_2.11/).

**Comment by bvenners** [comment]
@tindzk That's true! Closing.
